clients/iota-go/iotaclient/api_coin_query.go

```diff
@@ -29,7 +29,7 @@ func (c *Client) GetAllCoins(ctx context.Context, req GetAllCoinsRequest) (*iota
 
 type GetBalanceRequest struct {
 	Owner    *iotago.Address
-	CoinType iotago.ObjectType // optional
```

Comment:
Why was this ObjectType removed? 

Wouldn't it make sense to keep a specific type for CoinTypes coming from the API, just to indicate it properly?

Reply:
I replaced the old type `ObjectType = string` with a more strict `ObjectType struct {s string}` to avoid subtle bugs where the string contains an invalid or non-canonical format.

Here I just replaced the argument type with `string`, so it keeps the functionality. I didn't remove it.